src/clj/game/core/drawing.clj

```diff
@@ -27,45 +28,56 @@
-                   (swap! state update-in [side :register :drawn-this-turn] (fnil #(+ % draws-after-prevent) 0))
-                   (swap! state update-in [:stats side :gain :card] (fnil + 0) n)
-                   (swap! state update-in [:bonus] dissoc :draw)
-                   (if (and (not suppress-event) (pos? deck-count))
-                     (wait-for
-                       (trigger-event-sync state side (if (= side :corp) :corp-draw :runner-draw) draws-after-prevent)
-                       (let [eid (make-result eid drawn)]
-                         (trigger-event-sync state side eid (if (= side :corp) :post-corp-draw :post-runner-draw) draws-after-prevent)))
-                     (effect-completed state side eid))
-                   (when (safe-zero? (remaining-draws state side))
-                     (prevent-draw state side))))
-               (when (< draws-after-prevent draws-wanted)
-                 (let [prevented (- draws-wanted draws-after-prevent)]
-                   (system-msg state (other-side side) (str "prevents "
-                                                            (quantify prevented "card")
-                                                            " from being drawn"))))))))
-
+   (trigger-event state side (if (= side :corp) :pre-corp-draw :pre-runner-draw) n)
+   (let [n (+ n (get-in @state [:bonus :draw] 0))
+         draws-wanted n
+         active-player (get-in @state [:active-player])
+         draws-after-prevent (if (and (= side active-player) (get-in @state [active-player :register :max-draw]))
+                               (min n (remaining-draws state side))
+                               n)
+         deck-count (count (get-in @state [side :deck]))]
+     (when (and (= side :corp) (<= deck-count draws-after-prevent))
+       (win-decked state))
+     (when (< draws-after-prevent draws-wanted)
+       (let [prevented (- draws-wanted draws-after-prevent)]
+         (system-msg state (other-side side)
+                     (str "prevents " (quantify prevented "card") " from being drawn"))))
+     (if (or (and (= side active-player)
+                  (get-in @state [side :register :cannot-draw]))
+             (not (pos? draws-after-prevent))
+             (not (pos? deck-count)))
+       (effect-completed state side eid)
+       (let [to-draw (take draws-after-prevent (get-in @state [side :deck]))
+             drawn (mapv #(move state side % :hand) to-draw)
+             drawn-count (count drawn)]
+         (swap! state update-in [side :register :drawn-this-turn] (fnil #(+ % drawn-count) 0))
+         (swap! state update-in [:stats side :gain :card] (fnil + 0) n)
+         (swap! state update :bonus dissoc :draw)
+         (if suppress-event
+           (effect-completed state side eid)
+           (let [draw-event (if (= side :corp) :corp-draw :runner-draw)]
+             (swap! state assoc-in [side :register :currently-drawing] drawn)
```

Comment:
We should `conj` `drawn` to `:currently-drawing` here so we don't overwrite any pending draws.

Reply:
I don't think its's possible to draw during the resolution of a draw.